### PR TITLE
Add docs on 0.8 -> 0.9 compatibility

### PR DESCRIPTION
**Pull request by anyasabo**
Relates to https://github.com/elastic/cloud-on-k8s/issues/1242

I wasn't sure if there was a different place we also wanted to add these. Maybe also in `release-notes/0.9.0.asciidoc`?

### CODE REVIEW

**Comment by pebrc** [comment]
Maybe in a separate file/chapter like https://www.elastic.co/guide/en/elasticsearch/reference/current/breaking-changes-7.2.html

File: docs/release-notes/highlights-0.9.0.asciidoc [hunk 1]
```diff
@@ -29,4 +29,7 @@ this allows you to configure your own init containers to run code before Elastic
 Or you can use the Pod template to mount additional volumes with configuration files, for example to configure SAML or synonym files,
 or to specify `imagePullSecrets` for your private Docker registry.
 
+[float]
+=== Breaking Changes
 
+Existing resources (Elasticsearch and Kibana) that were created with a 0.8.x version of the operator are *not* compatible with this release. They will automatically have an annotation added to them indicating the controller version, and then skipped. They will continue to function, but any changes made to them will not be processed by the 0.9.x version of the operator. To upgrade existing clusters, please https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-snapshots.html[perform a snapshot] of the existing cluster and restore them into a new cluster created by the 0.9.x version of the operator.
```
Comment (pebrc) [inline docs/release-notes/highlights-0.9.0.asciidoc]: Reindex from remote is also an option https://www.elastic.co/guide/en/elasticsearch/reference/current/reindex-upgrade-remote.html

Comment (anyasabo) [reply docs/release-notes/highlights-0.9.0.asciidoc]: Ah good point

**Comment by sebgl** [comment]
I pushed some commits on your branch to document (and fix) https://github.com/elastic/cloud-on-k8s/issues/1384. Sorry for stepping onto your PR, it was convenient 😄

**Pull request merged**